extensions/http/src/main/java/org/mule/extension/http/api/listener/HttpListenerProvider.java

```diff
@@ -136,37 +152,68 @@ public void initialise() throws InitialisationException
         }
 
         verifyConnectionsParameters();
-    }
 
-    @Override
-    public Server connect() throws ConnectionException
-    {
+        //TODO: MULE-9320 Define threading model for message sources in Mule 4 - Analyse whether this can be avoided
+        workerThreadingProfile = new MutableThreadingProfile(DEFAULT_THREADING_PROFILE);
+        workerThreadingProfile.setMaxThreadsActive(DEFAULT_MAX_THREADS);
+
         HttpServerConfiguration serverConfiguration = new HttpServerConfiguration.Builder()
                 .setHost(host)
                 .setPort(port)
                 .setTlsContextFactory(tlsContext)
                 .setUsePersistentConnections(usePersistentConnections)
                 .setConnectionIdleTimeout(connectionIdleTimeout)
-                .setOwnerName(configName)
+                .setWorkManagerSource(createWorkManagerSource(workManager))
                 .build();
-        Server server = connectionManager.create(serverConfiguration);
+        try
+        {
+            server = connectionManager.create(serverConfiguration);
+        }
+        catch (ConnectionException e)
+        {
+            throw new InitialisationException(createStaticMessage("Could not create HTTP server"), this);
+        }
+    }
 
+    @Override
+    public void start() throws MuleException
+    {
         try
         {
+            workManager = createWorkManager(configName);
+            workManager.start();
             server.start();
         }
         catch (IOException e)
         {
-            throw new ConnectionException("Could not start server", e);
+            throw new DefaultMuleException("Could not start HTTP server", e);
+        }
+    }
+
+    @Override
+    public void stop() throws MuleException
+    {
+        try
+        {
+            workManager.dispose();
+        }
+        finally
+        {
+            workManager = null;
         }
+        server.stop();
```

Comment:
shouldn't the server be stopped before the workManager so that you avoid situations in which you still accept connections but there're no workers to handle them? Also I think that one should be stopped on the other's finally. As it is now, if `workManager.dispose()` were to fail the server would leak (I know that dispose() doesn't throw exceptions but a RTE can happen unexpectedly)